covid_moonshot/analysis/plots.py

```diff
@@ -337,10 +374,16 @@ def save_plot(path: str, name: str, file_format: str):
 
     os.makedirs(path, exist_ok=True)
 
-    plt.figure()
-    yield
-    plt.tight_layout()
-    plt.savefig(os.path.join(path, os.extsep.join([name, file_format])))
+    if table:
```

Comment:
If there's not much overlap between the branches with `table == True` and `False`, I'd suggest adding a new function instead (e.g. `save_table`).

Reply:
Would `save_table` need a `@contextgenerator` decorator?

Reply:
To use it the way it currently is (as `with save_table ...`), yes. Would something like the following work? (from example [here](https://matplotlib.org/3.1.0/gallery/misc/multipage_pdf.html))

``` python
@contextmanager
def save_table_pdf(path: str, name: str):

    # Make sure the directory exists
    import os

    os.makedirs(path, exist_ok=True)
    file_name = os.path.join(path, os.extsep.join([name, "pdf"]))

    with PdfPages(file_name) as pdf_plt:
        yield
        pdf_plt.savefig()
```

Reply:
Ok that's great, thanks!